tree/ntuple/v7/src/RNTupleDescriptor.cxx

```diff
@@ -844,6 +846,27 @@ ROOT::Experimental::Internal::RNTupleDescriptorBuilder::AddColumn(RColumnDescrip
    return RResult<void>::Success();
 }
 
+ROOT::Experimental::RResult<void>
+ROOT::Experimental::Internal::RNTupleDescriptorBuilder::AttachColumn(DescriptorId_t fieldId,
+                                                                     DescriptorId_t logicalColumnId)
```

Comment:
Why do we need a second function for this (that we can potentially forget to call), can we just do this at the end of `AddColumn` which should already have all necessary information?

(drive-by comment: can we de-duplicate the two `AddColumn` implementations? They seem to be largely doing the same, modulo error checking, in sometimes different order...)

Reply:
(ah, I see you do this in the third commit; please consider squashing its changes into the first commit)